### ISSUE TITLE
New way to connect to the other nodes

### ISSUE THREAD

**Issue by spikeekips** [bosnet/sebak#659]
~~Originally `/connect` was devised for connecting nodes each other, so `/connect` will send the node information and the the other node checks it. The missing thing is if the node information is updated like it's endpoint or it's validators, the other nodes should update also validators information. With updating validator information, the node will broadcast to the updated endpoint without restarting.~~

~~* To validate the data from `/connect`, the message of `/connect` should be signed by node's key~~
~~* If validated and some informations are changed, the received node should update it's validator.~~

Currently the node connects to the other validator nodes,

* Node must know the endpoint and public address of each nodes.
* Node must check the connection is made with each nodes.

This way is simple and easy to understand, but this may have some inefficiency and weaknesses for node operation,

* There is not validation process, the connected node is **really** the node.
* If the endpoint of some node is changed, the other node must restart with new endpoint.

---
To make this way to be better, I suggest new way, that is,

* We need the new 2 message, `ConnectMessage` and `ConnectedMessage`
  * `ConnectMessage` and `ConnectedMessage` have the signature, signed by sender node.
  * Node sends `ConnectMessage` and gets the `ConnectedMessage` as response.

* Node must know all the public addresses of the other nodes and **at least one of endpoints** of the other nodes.
* When node starts, node sends `ConnectMessage` to the other nodes, which the node knows their endpoints.

* The other nodes can verify the received `ConnectMessage`.
* The other nodes responses the `ConnectedMessage`, which has the nodes of the received node.
* If in responded `ConnectedMessage` there are the nodes, which is not connected, the node sends `ConnectMessage` to the unconnected nodes
* Basically `ConnectMessage` has also `node.NodeInfo`, which already contains published `Endpoint` of node.

This new way can make,

* The validation can be possible whether the connected node is what I want to connect.
* If the endpoint of node is changed, the other node easily know the new endpoint, so without shutdown, all the operation of node will be fine.
